checker/src/main/java/org/checkerframework/checker/optional/OptionalVisitor.java

```diff
@@ -328,9 +360,120 @@ public void handleConditionalStatementIsPresentGet(IfTree tree) {
   public Void visitMethodInvocation(MethodInvocationTree tree, Void p) {
     handleCreationElimination(tree);
     handleNestedOptionalCreation(tree);
+    updateMethodNamesToEnclosingMethods(tree);
     return super.visitMethodInvocation(tree, p);
   }
 
+  /**
+   * Updates {@link methodNamesToEnclosingMethods} given a method invocation.
+   *
+   * <p>Check whether the method is in the set of methods that must be checked by the Non-Empty
+   * checker whenever a method invocation is encountered. If the method is in the set, the method
+   * that immediately encloses the method invocation should also be added to the set of methods to
+   * be checked by the Non-Empty Checker.
+   *
+   * <p>This ensures that the <i>clients</i> of any methods that must be checked by the Non-Empty
+   * Checker (i.e., methods that have preconditions related to the Non-Empty type system) are
+   * included in the set of methods to check.
+   *
+   * @param tree a method invocation tree
+   */
+  private void updateMethodNamesToEnclosingMethods(MethodInvocationTree tree) {
+    String invokedMethodName = tree.getMethodSelect().toString();
+    MethodTree enclosingMethod = TreePathUtil.enclosingMethod(this.getCurrentPath());
+    if (enclosingMethod != null) {
+      Set<String> namesOfMethodsForNonEmptyChecker =
```

Comment:
This needs a comment noting that using the method name is a safe but imprecise approximation, since it will catch all overloads in its dragnet.  Or, change it to use the full signature or some other unique representation.

The same goes for `calleesToCallers`/`methodNamesToEnclosingMethods` and perhaps elsewhere.